### ISSUE TITLE
Doesn't seem to work with https git remotes

### ISSUE THREAD

**Issue by andyjeffries** [qinezh/vscode-gitlink#41]
Settings.json:

```
{
  "gitlink.defaultRemote": "origin",
}
```

Git remote output:

```
origin  git@github.com:repo-owner/repo-name.git (fetch)
origin  git@github.com:repo-owner/repo-name.git (push)
staging https://git.heroku.com/repo-name-staging.git (fetch)
staging https://git.heroku.com/repo-name-staging.git (push)
```

When I try to do "Goto Online link I get":

```
Can't parse https://git.heroku.com/repo-name-staging.git with Default rules
```

I'm running v0.8.1

**Comment by qinezh** [comment]
@andyjeffries it looks like a **self-host repo** in heroku.com for the staging, which is not supported by now.

**Comment by andyjeffries** [comment]
I don't understand what you mean? That's how people deploy to Heroku - by adding a git remote.

Also, given that my `defaultRemote` is set to `origin` why is it even caring about the Heroku remote?

**Comment by andyjeffries** [comment]
So, sorry when I said "doesn't seem to work with https git remotes", I didn't mean that I couldn't use it - but that the entire plugin doesn't work if you have one, even if you explicitly set it not to use that remote but a regular GitHub remote.

**Comment by qinezh** [comment]
@andyjeffries It should work for GitHub no matter via ssh or https. I have just verified from my side, would you please double check?
```bash
$ git remote -v
https   https://github.com/qinezh/vscode-gitlink.git (fetch)
https   https://github.com/qinezh/vscode-gitlink.git (push)
origin  git@github.com:qinezh/vscode-gitlink (fetch)
origin  git@github.com:qinezh/vscode-gitlink (push)
```

**Comment by andyjeffries** [comment]
OK, so I checked out your repo:

```
$ git clone git@github.com:qinezh/vscode-gitlink
```

Then I edited `.git/config` to contain:

```
[remote "https"]
	url = https://github.com/qinezh/vscode-gitlink.git
	fetch = +refs/heads/*:refs/remotes/origin/*
[remote "origin"]
	url = git@github.com:qinezh/vscode-gitlink
	fetch = +refs/heads/*:refs/remotes/origin/*
```

And it worked fine when I did "Copy online link".

Then I changed it to be:

```
[remote "https"]
	url = https://git.heroku.com/repo-name-staging.git
	fetch = +refs/heads/*:refs/remotes/origin/*
[remote "origin"]
	url = git@github.com:qinezh/vscode-gitlink
	fetch = +refs/heads/*:refs/remotes/origin/*
```

And it breaks. I then did:

```
$ mkdir .vscode
$ echo '{"gitlink.defaultRemote": "origin",}' > .vscode/settings.json
```

That should force it to ignore the "https" remote and use the "origin" one, right?

Same issue.

**Comment by andyjeffries** [comment]
It looks like it doesn't like "https://HOSTNAME/repo.git" only "https://HOSTNAME/ACCOUNT/repo.git", without that folder it breaks.

However (and this is most important) with the setting for `defaultRemote` as documented set to a `git@github.com` remote, it shouldn't even be trying to **look at or parse** the https one, it should just completely ignore it.

**Comment by qinezh** [comment]
Yeah, that would be a good point. Marked it as bug. Thank you for reporting it.

**Comment by andyjeffries** [comment]
No worries. Never looked at Typescript before, but I'm trying to have a look at raising a PR for git-urls. It may end up being ignored or implementing differently, but at least I'm not just saying "here's a bug, please fix it" but trying to help out at the same time :-)

**Comment by andyjeffries** [comment]
https://github.com/qinezh/git-urls/pull/28

**Comment by qinezh** [comment]
@andyjeffries this issue should be resolved in latest release 1.0.1, please take a try 😄

**Comment by andyjeffries** [comment]
Yep, working perfectly for me now. Many thanks!
